src/gui/guiEngine.cpp

```diff
@@ -202,6 +204,57 @@ GUIEngine::GUIEngine(JoystickController *joystick,
 	m_menu.reset();
 }
 
+
+/******************************************************************************/
+std::string findLocaleFileInMods(const std::string &path, const std::string &filename)
+{
+	std::vector<ModSpec> mods = flattenMods(getModsInPath(path, "root", true));
+
+	for (const auto &mod : mods) {
+		std::string ret = mod.path + DIR_DELIM "locale" DIR_DELIM + filename;
+		if (fs::PathExists(ret)) {
+			return ret;
+		}
+	}
+
+	return "";
+}
+
+/******************************************************************************/
+Translations *GUIEngine::getContentTranslations(const std::string &path,
+		const std::string &domain, const std::string &lang_code)
+{
+	if (domain.empty() || lang_code.empty())
+		return nullptr;
+
+	std::string filename = domain + "." + lang_code + ".tr";
+	std::string key = path + DIR_DELIM "locale" DIR_DELIM + filename;
+
+	auto it = translations.find(key);
+	if (it != translations.end())
+		return &it->second; // Already loaded
+
+	std::string trans_path = key;
+	ContentType type = getContentType(path);
+	if (type == ContentType::GAME)
+		trans_path = findLocaleFileInMods(path + DIR_DELIM "mods" DIR_DELIM, filename);
+	else if (type == ContentType::MODPACK)
+		trans_path = findLocaleFileInMods(path, filename);
+
```

Comment:
What about mod translations? They are documented in lua_api.md but I do not see any handling here.

Reply:
mods always have locale in one folder - we don't to recursively search for them

Reply:
Oh! I totally overlooked the fact that `trans_path` defaults to `key`. Makes sense :+1: